src/databricks/sql/client.py

```diff
@@ -196,9 +197,11 @@ def read(self) -> Optional[OAuthToken]:
             **kwargs,
         )
 
-        self._session_handle = self.thrift_backend.open_session(
+        self._open_session_resp = self.thrift_backend.open_session(
             session_configuration, catalog, schema
         )
+        self._session_handle = self._open_session_resp.sessionHandle
```

Comment:
The change on line 199 and 203 effectively changes the type of `self._session_handle` from an `openSessionResponse` to an actual `sessionHandle` object. Have you confirmed that this works in the other places where `self._session_handle` is used?

I suspect that it _does_ work because both the old type (`openSessionResponse`) and the new type (`sessionHandle`) have a property called `serverProtocolVersion`. But if anywhere else in the code we check for a property other than `self._session_handle.serverProtocolVersion` and this property is not present on the `sessionHandle` type, it will fail.

My guess is that the original name for this property (`self._session_handle`) was imprecise and your change is making the name more accurately match its contents 👍

Reply:
No, the original behavior was to return the same type of handle.

Reply:
Thanks for clarifying. I think somehow I missed the final clause of line 203 🤦